download.py

```diff
@@ -0,0 +1,185 @@
+            for idx in range(index, range_index):
+                url = f"{baseurl}images/part-{idx:06}.zip"
+                loop_file_path = f"{output}part-{index:06}.zip"
+# It's trying to download the file, and if it encounters an HTTPError, it prints the error.
+                try:
+                    urlretrieve(url, loop_file_path)
+                except HTTPError as e:
+                    print(f"HTTPError downloading file: {url} - {e}")
+                files_to_unzip.append(loop_file_path)
+# It's writing the url of the file to a manifest file.
+                with open("manifest.txt", "a") as f:
+                    f.write(f"{url}")
+                time.sleep(0.1)
+                bar()
+# It's checking if the user wants to unzip the files, and if they do, it returns a list of files to
+# unzip. It would be a bad idea to put these together as the process is already lengthy.
+    if unzip and len(files_to_unzip) > 0:
+        return files_to_unzip
+
+
+def unzip(file: str):
+    """
+    > This function takes a zip file as an argument and unpacks it
+    
+    :param file: str
+    :type file: str
+    :return: The file name without the .zip extension
+    """
+    shutil.unpack_archive(file)
+    return f"File: {file.replace('.zip', '')} has been unzipped"
+
+
+def unzip_all(files: list):
+    """
+    > Unzip all files in a list of files
+    
+    :param files: list
+    :type files: list
+    """
+    with alive_bar(len(files), title="Unzipping files") as bar:
+        for file in files:
+            unzip(file)
+            time.sleep(0.1)
+            bar()
+
+
+def main(index=None, range_max=None, output=None, unzip=None):
+    """
+    `main` is a function that takes in an index, a range_max, an output, and an unzip, and if the user
+    confirms that they have enough space, it downloads the files from the index to the output, and if
+    unzip is true, it unzips them
+    
+    :param index: The index of the file you want to download
+    :param range_max: The number of files to download
+    :param output: The directory to download the files to
+    :param unzip: If you want to unzip the files after downloading them, set this to True
+    :return: A list of files that have been downloaded
+    """
+    confirmation = input("Do you have at least 1.7Tb free: (y/n)")
+    if confirmation != "y":
+        return
+    if index and range_max:
+        files = download(index, range_max, output)
+        if unzip:
+            unzip_all(files)
+    elif index:
+        download(index, output=output)
+    else:
+        print("No index provided")
+
+
+# This is a common pattern in Python. It allows you to run the main function of your script by
+# running the script through the interpreter. It also allows you to import the script into the
+# interpreter
+# without automatically running the main function.
+if __name__ == "__main__":
+    main(index, range_max, output, unzip)
+
+"""
```

Comment:
Thanks for acknowledgements! Suggest removing it from this file though 😄

Reply:
Can do :D